io_stream.c

```diff
@@ -141,6 +185,14 @@ void* php_mongo_io_stream_connect(mongo_con_manager *manager, mongo_server_def *
 				return NULL;
 			}
 		} else {
+#if PHP_VERSION_ID < 50600
+			if (php_mongo_verify_hostname(manager, stream, server TSRMLS_CC) == FAILURE) {
```

Comment:
This is prone to returning `FAILURE` even if the user isn't requesting host name validation via the `peer_certificate` SSL stream context option.